File: spark-events-ui/src/app/model/ExecutorInfo.ts

/**
 * 
 */
export class LogUrls {
  stdout: string;
  stderr: string;

  constructor(stdout: string, stderr: string) {
    this.stdout = stdout;
    this.stderr = stderr;
  }

  static fromJson(src: any): LogUrls {
    let stdout = <string> src['stdout'];
    let stderr = <string> src['stderr'];
    return new LogUrls(stdout, stderr);
  }

  static createDefault(): LogUrls {
    return new LogUrls('', '');
  }

}

/**
 * 
 */
export class ExecutorInfo {

  // @JsonProperty("Host")
  host: string;

  // @JsonProperty("Total Cores")
  totalCores: number;

  // @JsonProperty("Log Urls")
  logUrls: LogUrls;

  // @JsonProperty("Attributes")
  attributes: Map<string,string>;

  // @JsonProperty("Resources")
  resources: Map<string,string>;

  // @JsonProperty("Resource Profile Id")
  resourceProfileId: number;

  constructor(host: string, totalCores: number,
      logUrls: LogUrls, attributes: Map<string,string>, resources: Map<string,string>, resourceProfileId: number) {
    this.host = host;
    this.totalCores = totalCores;
    this.logUrls = logUrls;
    this.attributes = attributes;
    this.resources = resources;
    this.resourceProfileId = resourceProfileId;
  }

  static fromJson(src: any): ExecutorInfo {
    let host = <string> src['Host'];
    let totalCores = src['Total Cores'];
    let logUrls = LogUrls.fromJson(src['Log Urls']);
    let attributesObj = src['Attributes'];
    let attributes = (attributesObj)? new Map(Object.entries(attributesObj)) : new Map();
    let resourcesObj = src['Resources'];
    let resources = (resourcesObj)? new Map(Object.entries(resourcesObj)) : new Map();
    let resourceProfileId = <number> src['Resource Profile Id'];
    return new ExecutorInfo(host, totalCores, logUrls, attributes, resources, resourceProfileId);
  }

  static createDefault(): ExecutorInfo {
    let logUrls = LogUrls.createDefault();
    return new ExecutorInfo('', 0, logUrls, new Map(), new Map(), 0);
  }
}

File: spark-event-gateway-srv/src/main/java/fr/an/spark/gateway/eventlog/model/ExecutorInfo.java
package fr.an.spark.gateway.eventlog.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.HashMap;
import java.util.Map;

/**
 * https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/cluster/ExecutorInfo.scala#L28
 */
public class ExecutorInfo {

    @JsonProperty("Host")
    public String host;

    @JsonProperty("Total Cores")
    public int totalCores;

//    public static class LogUrls {
//        public String logs;
//        public String metrics;
//        public String stdout;
//        public String stderr;
//    }
    
    @JsonProperty("Log Urls")
    public Map<String,String> logUrls;

    @JsonProperty("Attributes")
    public Map<String,String> attributes;

    @JsonProperty("Resources")
    public Map<String,ResourceInformation> resourcesInfo;

    @JsonProperty("Resource Profile Id")
    public int resourceProfileId;

    public Long registrationTime;

    public Long requestTime;

    // TODO
    public static class ResourceInformation extends HashMap<String,Object> {
    }

}
